azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -1008,6 +1067,32 @@ func resourceLinuxVirtualMachineUpdate(d *schema.ResourceData, meta interface{})
 		log.Printf("[DEBUG] Updated Linux Virtual Machine %q (Resource Group %q).", id.Name, id.ResourceGroup)
 	}
 
+	// TODO Beta flag here
+	if true {
+		// TODO - delete removed disks if feature flag set
+		if deleteRemovedDisks && len(dataDisksToDeleted) > 0 {
+			for _, v := range dataDisksToDeleted {
+				if v.ManagedDisk != nil && v.ManagedDisk.ID != nil {
```

Comment:
Can we skip nil checks here/

Reply:
These objects have been populated from an API response and will not have been checked for this item's validity before this point, so I think this needs to stay.